ccswbs/gus
# Testimonials

# Summary of changes
- Add Testimonial Slider component to Basic page widgets. It allows users to add testimonials either individually or by term (testimonial type, tags, program, degrees, unit, specialisations, unit).

## Frontend
- Create testimonialSlider component for paragraph widget
- Adapt Testimonial component to work with either program page or testimonialSlider widget

### Code changes
- Add TestimonialNodeFragment to Program page
- Fix console warnings and indents on index page and OVC custom footer (adding key to maps)
- Remove TestimonialTags file (worked code into Testimonial)

## Backend
- *Multidev:* https://testimonial-chug.pantheonsite.io
- Add Testimonial Slider paragraph as basic page widget
  - Includes title for heading
  - Allows users to add individual testimonial nodes or testimonials by tag (e.g. Faculty)
- Update (existing) Testimonial content type
  - Add field for testimonial person name (as opposed to using Title - this allows the same person to give multiple testimonials)
  - Use separate field for referencing Testimonial Type (e.g. Student, Faculty, etc.)
  - Re-add field_home_profile field (it looks like it was accidentally removed at some point)

# Test Plan
- For expected rendering, view https://build-a4fdbc45-19a9-4ad3-89bb-6529c553c261.gtsb.io/testimonial-slider-test
- Try adding a new testimonial:
  - Log into http://testimonial-chug.pantheonsite.io
  - Add a new testimonial node
  - On a basic page, add a testimonial slider widget
  - Try adding some testimonials (both individually and by tag)
- Check the structure:
  - View Testimonial Slider paragraph
  - View the testimonial node
  - Check permissions on Testimonial Slider component
- Check index page and confirm there are no console warnings
- Check an OVC page and confirm there are no console warnings relating to the footer
